src/ipc/ipc4/handler.c

```diff
@@ -875,6 +875,9 @@ static int ipc4_module_process_dx(struct ipc4_message_request *ipc4)
 			return IPC4_BUSY;
 		}
 
+#if defined(CONFIG_PM)
+		ipc_get()->task_mask |= IPC_TASK_POWERDOWN;
```

Comment:
if this is going to be set here, then maybe we can also remove the redundant code in https://github.com/thesofproject/sof/blob/0664718a5896abe1e50191218cb80abb7171ad2d/src/platform/intel/cavs/platform.c#L633